shared/src/summarize_result.js

```diff
@@ -63,50 +66,50 @@ async function setup() {
       }
 
       summaryResultElement.style.display = '';
-      summaryResultElement.innerHTML = data.summary.replaceAll(/\n/g, '<br />');
+      summaryResultElement.innerText = summaryTextContents;
     }
   });
 
   async function requestPageSummary() {
+    const hasTabAccess = await browser.permissions.contains({
+      permissions: ['activeTab'],
+    });
+
+    if (!hasTabAccess) {
+      summaryResultElement.style.display = '';
+      summaryResultElement.classList.add('error');
+      summaryResultElement.innerText =
+        "You can't summarize without allowing access to the currently active tab.";
+      return;
```

Comment:
This is just a "safety check" in case the shortcut is set and called before permissions are given (can't request them from here _easily_ since it needs a button click, as an example -- it seemed unnecessary to rebuild that UI here, for this case, as it already exists in the main popup).